File: CPP/101. 对称二叉树.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        if(root == NULL) return true;
        return isSameTree(root->left, root->right);
    }

    bool isSameTree(TreeNode* left, TreeNode* right){
        if(left == NULL && right == NULL) return true;
        if(left == NULL || right == NULL) return false;
        if(left->val == right->val) return isSameTree(left->left, right->right) && isSameTree(left->right, right->left);
        return false;
    }
};

File: Java/101.对称二叉树.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null) return true;
        return isRightFuc(root.left, root.right);
        }

    public static boolean isRightFuc(TreeNode l, TreeNode r){
        if(l == null && r == null) return true;
        if(l == null || r == null) return false;
        if(l.val == r.val){
            return isRightFuc(l.left,r.right) && isRightFuc(l.right, r.left); 
        }
        return false;
    }
}

    
